src/main/java/org/jenkinsci/plugins/github_branch_source/BranchOrTagAgeDiscoveryTrait.java

```diff
@@ -0,0 +1,245 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.github_branch_source;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMHeadCategory;
+import jenkins.scm.api.SCMHeadOrigin;
+import jenkins.scm.api.SCMRevision;
+import jenkins.scm.api.SCMSource;
+import jenkins.scm.api.mixin.TagSCMHead;
+import jenkins.scm.api.trait.SCMHeadAuthority;
+import jenkins.scm.api.trait.SCMHeadAuthorityDescriptor;
+import jenkins.scm.api.trait.SCMHeadFilter;
+import jenkins.scm.api.trait.SCMSourceContext;
+import jenkins.scm.api.trait.SCMSourceRequest;
+import jenkins.scm.api.trait.SCMSourceTrait;
+import jenkins.scm.api.trait.SCMSourceTraitDescriptor;
+import jenkins.scm.impl.trait.Discovery;
+import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.github.GHBranch;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * A {@link Discovery} trait for GitHub that will discover branches with a certain age on the repository.
+ *
+ * @since 2.2.0
+ */
+public class BranchOrTagAgeDiscoveryTrait extends SCMSourceTrait {
+    private static final Logger LOGGER = Logger.getLogger(BranchOrTagAgeDiscoveryTrait.class.getName());
+    
+    /**
+     * The maximum age (in days) of branches or tags.
+     */
+    private int maxDaysOld = 0;
+
+    /**
+     * Constructor for stapler.
+     *
+     * @param maxDaysOld the maximum amount of days a branch or tag can be old.
+     */
+    @DataBoundConstructor
+    public BranchOrTagAgeDiscoveryTrait(int maxDaysOld) {
+        this.maxDaysOld = maxDaysOld;
+    }
+
+    /**
+     * Returns the maximum days a branch or tag can be old.
+     *
+     * @return the maximum days a branch or tag can be old.
+     */
+    public int getMaxDaysOld() {
+        return maxDaysOld;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext(SCMSourceContext<?, ?> context) {
+        GitHubSCMSourceContext ctx = (GitHubSCMSourceContext) context;
+        ctx.withAuthority(new BranchOrTagAgeSCMHeadAuthority());
```

Comment:
What does this do?

Reply:
As far as I could tell it is a way to activate trust for the head and request that is shared to other authorities (according to [this](https://github.com/jenkinsci/scm-api-plugin/blob/master/src/main/java/jenkins/scm/api/trait/SCMHeadAuthority.java#L41-L48)). But you are right, it should probably take `SCMHead` in the Authorities' `checkTrusted` method since we don't just accept branches but also . I will update it.

Reply:
Let me know if it is not needed from your perspective and I will remove it.